### ISSUE TITLE
Wrong click position when the window is attached to the top edge

### ISSUE THREAD

**Issue by oeway** [Xpra-org/xpra-html5#77]
Hi, I encountered a bug with the wrong clicking position in a popup menu, it happens when I have a window attached to the top edge.

In the GIF below, you will see that when I open the application, it automatically attach to the top edge, and the menu was also automatically attached which cause the wrong clicking position. And if I drag the window in a lower position, the popup menu works as expected.
![icouZa2FPC](https://user-images.githubusercontent.com/478667/127225786-23356cc6-4182-49c8-b744-b96eb30b3c84.gif)

You can reproduce the issue also by the following two steps: 
 1. Click https://mybinder.org/v2/gh/imjoy-team/imjoy-hybrid-desktop/xpra-bug-report?urlpath=desktop and wait for some minutes for the docker container to start
 2. In the terminal window run `./Fiji.app/ImageJ-linux64`
 3. Click the File menu.

**Comment by totaam** [comment]
What distribution are you using? What xpra version?
Have you tried switching to `Xvfb` instead of `Xdummy`? (or vice versa)

My guess is that the virtual screen is bigger than your browser window and the application requests a specific absolute position on screen, which ends up being off-screen.

**Comment by oeway** [comment]
Here are the version numbers:
```
xpra v4.2.1-r0
xpra html client v4.3
```

I installed Xvfb and started xpra with these arguments:
```
 'xpra', 'start',  '--bind-tcp=0.0.0.0:{port}',  '--html=on',  '--start=lxterminal',  '--daemon=no',  ":200"
```
So this means I used `Xvfb`? Not sure how to switch to Xdummy, should I just add `xvfb=Xorg`?

Another observation is that when I drag the window close to the edge, it automatically stick to the edge, is there a way to disable that automatic sticking? And when I change the browser window size, the startup logo is always in the center of the screen which to me seems like it does adapt to my browser window size.

**Comment by oeway** [comment]
Hi, I just dug into the code a bit without fully understand it, but if I disable the `XpraWindow.prototype.ensure_visible` function (add a return in the beginning) and the everything works normally. 

It seems to me that there is some issue with the visibility detection logic. Essentially when the windows and popup menus are actually visible, it still try to move it into a wrong position without synchronizing the mouse clicking position. Does that make sense?

**Comment by totaam** [comment]
@oeway you haven't specified your distribution.

> So this means I used `Xvfb`?

No, installing `Xvfb` alone doesn't change the configuration.
See `/etc/xpra/conf.d/55_server_x11.conf`.

> Another observation is that when I drag the window close to the edge, it automatically stick to the edge, is there a way to disable that automatic sticking?

I have no idea how, but it should be possible.

> .. the startup logo is always in the center of the screen which to me seems like it does adapt to my browser window size

How well it resizes depends on whether you use `Xvfb` or `Xdummy`.

> Essentially when the windows and popup menus are actually visible, it still try to move it into a wrong position without synchronizing the mouse clicking position...

> set the `min_visible=0`

This effectively skips the `ensure_visible` function.
I think that somehow the server window geometry and html5 client geometry are not in sync.
From what I am seeing, `send_configure_window` should be called when `ensure_visible` returns `false` and this should keep things in sync.

**Comment by oeway** [comment]
> you haven't specified your distribution.

Do you mean the distribution of linux? I mentioned that it's Ubuntu bionic. If helpful, here is the docker file I used: https://github.com/imjoy-team/imjoy-hybrid-desktop/blob/xpra-bug-report/binder/Dockerfile

> How well it resizes depends on whether you use Xvfb or Xdummy. 

I think I am using Xvfb, tried the options for Xdummy but it complains with an error.

> I think that somehow the server window geometry and html5 client geometry are not in sync.
From what I am seeing, send_configure_window should be called when ensure_visible returns false and this should keep things in sync.

Could you please try it yourself with the steps I provided in the description of this issue? It takes <1 minutes to start the server.

**Comment by totaam** [comment]
> I think I am using Xvfb

Yes, that's the default on Ubuntu.

I can't use a container for debugging or development. Surely there is an open-source application I can just install to reproduce the problem?

**Comment by oeway** [comment]
Sure, here it is
https://imagej.net/software/fiji/downloads

**Comment by totaam** [comment]
The fix above is an improvement, but I will try to find a better fix.

**Comment by oeway** [comment]
@totaam Great! Thanks a lot!

**Comment by totaam** [comment]
I could reproduce the problem simply with the xterm's menu (control+click) and reverting 9179e64f98cb5fee3a72a72c0104e0930c6e0fe1.
The correct fix is in 46ee8712ee1a6a8e1be9581b5ab5b36d0304d865.
The html5 client could be taught how to shift override-redirect windows when needed. This is not something that the server can do currently.
